src/material/paginator/paginator.ts

```diff
@@ -106,7 +106,7 @@ export interface MatPaginatorSelectConfig {
 export abstract class _MatPaginatorBase<
     O extends {
       pageSize?: number;
-      pageSizeOptions?: number[];
+      pageSizeOptions?: (number | {value: number; label: string})[];
```

Comment:
Maybe move `(number | {value: number; label: string})` into a separate type so we don't have to repeat it.